tests/DefaultSpec.hs

```diff
@@ -0,0 +1,40 @@
+-- |
+-- Module      : DefaultSpec
+-- Description : Ensure that deriving via (Default  a)newtype works
+-- License     : BSD-3-Clause
+--
+-- Maintainer  : generics@haskell.org
+-- Stability   : experimental
+-- Portability : non-portable
+--
+-- Tests DerivingVia on GHC versions 8.6 and above.
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 806
+{-# LANGUAGE DerivingVia #-}
+#endif
+
+module DefaultSpec where
+
+import Test.Hspec
+import Generics.Deriving
```

Comment:
You'll need to surround these `Generics.Deriving.*` imports with `#if __GLASGOW_HASKELL__ >= 806 ... #endif` to fix the Travis build.